src/sentry/integrations/slack/integration.py

```diff
@@ -189,6 +190,32 @@ def build_integration(self, state):
 
         return integration
 
+    def create_audit_log_entry(self, integration, organization, request, action, extra=None):
+        if action == "upgrade":
+            create_audit_entry(
+                request=request,
+                organization=organization,
+                target_object=integration.id,
+                event=AuditLogEntryEvent.INTEGRATION_UPGRADE,
+                data={"provider": integration.provider, "name": integration.name},
+            )
+
+            if extra and extra.get("extra_orgs"):
+                for org in Organization.objects.filter(slug__in=extra["extra_orgs"]):
+                    create_audit_entry(
+                        request=request,
+                        organization=org,
+                        target_object=integration.id,
+                        event=AuditLogEntryEvent.INTEGRATION_UPGRADE,
```

Comment:
@MeredithAnya Will it be possible which org actually did the upgrade based on the audit entries? Or do we have to look at logging to figure that out?

Reply:
@scefali hmmm probs just logging here.